packages/shared/useDateFormat/index.ts

```diff
@@ -16,25 +23,28 @@ export const formatDate = (date: Date, formatStr: string) => {
   const seconds = date.getSeconds()
   const milliseconds = date.getMilliseconds()
   const day = date.getDay()
-  const matches: Record<string, string | number> = {
-    YY: String(years).slice(-2),
-    YYYY: years,
-    M: month + 1,
-    MM: `${month + 1}`.padStart(2, '0'),
-    D: String(days),
-    DD: `${days}`.padStart(2, '0'),
-    H: String(hours),
-    HH: `${hours}`.padStart(2, '0'),
-    h: `${hours % 12 || 12}`.padStart(1, '0'),
-    hh: `${hours % 12 || 12}`.padStart(2, '0'),
-    m: String(minutes),
-    mm: `${minutes}`.padStart(2, '0'),
-    s: String(seconds),
-    ss: `${seconds}`.padStart(2, '0'),
-    SSS: `${milliseconds}`.padStart(3, '0'),
-    d: day,
+  const matches: Record<string, () => string | number> = {
```

Comment:
Changed to function so as not to unnecessarily execute toLocaleDateString and degrade performance.